public/javascripts/main.js

```diff
@@ -1111,33 +1120,75 @@ lichess.notifyApp = (function() {
       self.$nbOnline = self.$title.find('.online');
       self.$nobody = self.element.find("div.nobody");
       self.set(self.element.data('preload').split(','));
+      self.playings(self.element.data('playing').split(','));
+    },
+    _makeUser: function(name) {
+      return {
+        'name' : name
+      }
     },
     repaint: function() {
       this.users = lichess.unique(this.users.filter(function(u) {
-        return u !== '';
+        return u.name !== '';
       }));
       this.$nbOnline.text(this.users.length);
       this.$nobody.toggle(this.users.length === 0);
       this.$list.html(this.users.map(this._renderUser).join(""));
       $('body').trigger('lichess.content_loaded');
     },
     set: function(us) {
-      this.users = us || [];
+      this.users = us.map(this._makeUser) || [];
       this.repaint();
     },
-    enters: function(user) {
+    enters: function(userName) {
+      var user = this._makeUser(userName);
+
       this.users.push(user);
       this.repaint();
     },
-    leaves: function(user) {
+    leaves: function(userName) {
       this.users = this.users.filter(function(u) {
-        return u != user
+        return u.name != userName
       });
       this.repaint();
     },
+    _setPlaying: function(userName, playing) {
+
+        var user = this.users.find(function(u) {
```

Comment:
poor browser support for `find`, better use `filter` https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/find